src/api/app/models/staging/staged_requests.rb

```diff
@@ -84,6 +85,35 @@ def requests
     staging_workflow.unassigned_requests.where(number: request_numbers)
   end
 
+  def link_package(bs_request_action)
+    request = bs_request_action.bs_request
+
+    source_package = Package.get_by_project_and_name!(bs_request_action.source_project,
+                                                      bs_request_action.source_package)
+
+    # get the revision from the source package
+    source_rev = bs_request_action.source_rev
+
+    # find or create the package in the staging project
+    link_package = Package.find_or_create_by!(project: staging_project, name: bs_request_action.target_package)
+
+    # add or overwrite "_link", linking to the source project baserev
+    Backend::Api::Sources::Package.write_link(staging_project.name,
+                                              link_package.name,
+                                              User.session!, "<link project=\"#{source_package.project.name}\" rev=\"#{source_rev}\"></link>")
```

Comment:
You forgot the package here and we need to fixate the `vrev` - because the `source_rev` needs to be expanded `srcmd5` and quite some packages won't work if the start with revision 1.

Just as in the code you're replicating: https://github.com/openSUSE/openSUSE-release-tools/blob/master/osclib/stagingapi.py#L1132